packages/router/router.ts

```diff
@@ -1304,12 +1313,7 @@ export function createRouter(init: RouterInit): Router {
       state.fetchers.set(key, loadingFetcher);
       updateState({ fetchers: new Map(state.fetchers) });
 
-      let redirectNavigation: NavigationStates["Loading"] = {
-        state: "loading",
-        location: createLocation(state.location, actionResult.location),
-        ...submission,
-      };
-      await startRedirectNavigation(actionResult, redirectNavigation);
+      await startRedirectNavigation(state, actionResult);
       return;
```

Comment:
Since you're already in an `async` function, you can return right away without `await`ing the result

```suggestion
      return startRedirectNavigation(state, actionResult);
```